client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/HotRodOperation.java

```diff
@@ -122,6 +120,13 @@ public void exceptionCaught(Channel channel, Throwable cause) {
       }
    }
 
+   protected final boolean closeChannelForCause(Throwable t) {
+      // don't close the channel, server just sent an error, there's nothing wrong with the channel
+      return !((t instanceof HotRodClientException && ((HotRodClientException) t).isServerError())
```

Comment:
I believe the first part can be extracted to a method like
```java
   protected static boolean isServerError(Throwable t) {
      return t instanceof HotRodClientException && ((HotRodClientException) t).isServerError();
   }
```
https://github.com/infinispan/infinispan/blob/241f8d9ace0729106a5ef697fab5af098e919cde/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/RetryOnFailureOperation.java#L183

The line above can use the new method